### DISCUSSION TITLE
🐼 Panda v0.17.1

### DISCUSSION THREAD

**Discussion by anubra266** [chakra-ui/panda#1584] [Announcements]
To upgrade to v0.17.1:

```bash
pnpm up "@pandacss/*"
#or
yarn up '@pandacss/*'
```

### Fixed

- Fix persistent error that causes CI builds to fail due to PostCSS plugin emitting artifacts in the middle of a build
  process.
- Fix issue where conditions don't work in semantic tokens when using template literal syntax.
- Fix issue in reset styles where button does not inherit color style
- Fix issue where FileSystem writes cause intermittent errors in different build contexts (Vercel, Docker). This was
  solved by limiting the concurrency using the `p-limit` library
- Fix issue where using scale css property adds an additional 'px'
- Fix issue where styled objects are sometimes incorrectly merged, leading to extraneous classnames in the DOM

### Added

- Add `--host` and `--port` flags to studio.

### Changed

- Change `OmittedHTMLProps` to be empty when using `config.jsxStyleProps` as `minimal` or `none`
- Remove export types from jsx when no jsxFramework configuration
- Extract identifier values coming from an `EnumDeclaration` member

Example:

```ts
enum Color {
  Red = 'red.400',
  Blue = 'blue.400',
}

const className = css({ color: Color.Red, backgroundColor: Color['Blue'] })
```

- Use predefined interfaces instead of relying on automatic TS type inference or type aliases. This should result in
  snappier

This should fix issues with the generation of typescript declaration (`.d.ts`) files when using `@pandacss/xxx` packages
directly, such as:

```
src/config.ts(21,14): error TS2742: The inferred type of 'tokens' cannot be named without a reference to '../node_modules/@pandacss/types/src/shared'. This is likely not portable. A type annotation is necessa…
```

> These changes are only relevant if you are directly using **other** Panda `@pandacss/xxx` packages than the
> `@pandacss/dev`.

**Comment by olafurns7** [comment]
Just updated across our monorepo, from `0.16.0` to `0.17.1`, I get this error message when running `pnpm panda codegen --clean`


```bash
p panda codegen --clean
Panda in dev mode, running from source
node:internal/modules/cjs/loader:1051
  throw err;
  ^

Error: Cannot find module './src/cli-main'
Require stack:
- /Users/olafurns/Verkefni/dd-unity/node_modules/.pnpm/@pandacss+dev@0.17.1_@internationalized+date@3.5.0_@types+node@20.6.3_@types+react-dom@18.2.7_mc4eemvyusyy3hdoz4mcko437q/node_modules/@pandacss/dev/bin.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at requireSource (/Users/olafurns/Verkefni/dd-unity/node_modules/.pnpm/@pandacss+dev@0.17.1_@internationalized+date@3.5.0_@types+node@20.6.3_@types+react-dom@18.2.7_mc4eemvyusyy3hdoz4mcko437q/node_modules/@pandacss/dev/bin.js:20:19)
    at symlink (/Users/olafurns/Verkefni/dd-unity/node_modules/.pnpm/@pandacss+symlink@0.17.1/node_modules/@pandacss/symlink/src/index.js:83:3)
    at Object.<anonymous> (/Users/olafurns/Verkefni/dd-unity/node_modules/.pnpm/@pandacss+dev@0.17.1_@internationalized+date@3.5.0_@types+node@20.6.3_@types+react-dom@18.2.7_mc4eemvyusyy3hdoz4mcko437q/node_modules/@pandacss/dev/bin.js:17:22)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/olafurns/Verkefni/dd-unity/node_modules/.pnpm/@pandacss+dev@0.17.1_@internationalized+date@3.5.0_@types+node@20.6.3_@types+react-dom@18.2.7_mc4eemvyusyy3hdoz4mcko437q/node_modules/@pandacss/dev/bin.js'
  ]
}

Node.js v20.8.0

```

**Reply by ygorluiz** [reply]
@olafurns7 u did found a solution ?

**Reply by hrsh7th** [reply]
IMO, the fix is started.
https://github.com/chakra-ui/panda/commit/116c612e90003a6faff3fe094bb848db6776557d

**Reply by vitorGabr** [reply]
I still have this error, even when updated to version 0.17.1

**Reply by anubra266** [reply]
@vitorGabr check 0.17.2

**Reply by vitorGabr** [reply]
@anubra266  It worked
